### ISSUE TITLE
Verify release pipeline is working properly

### ISSUE THREAD

**Issue by eduardocerqueira** [apache/incubator-kie-issues#827]
in preparation for the Apache release, verify the release pipeline is working properly and make any adjustments if needed.
If possible run a test/dry-run for the pipeline.

**Comment by cimbalek** [comment]
First I tried to test this via my fork, but in fork there's too many things that need to be either altered or disabled, so we might run into false negatives (failures) caused by these changes rather than pipelines themselves. Therefore I decided to try branching via original jobs with version 9.9.9. My findings so far:

- Drools automated branching works without any issues.
- Optaplanner branching is blocked by #776 and the changes in configuration in the main branch will need to be commited manually

**Comment by cimbalek** [comment]
After creating new branches, the new jobs were successfully generated. Turned out that build and release pipeline are spanned across 10 repositories while each of them has its own specific implementation details. In order to not to risk releasing something we don't want to or sending notifications where it wouldn't be desired I decided, that I won't do the dry-run. It seemed to me too risky that such event would occur. But during visual inspection of jobs structure and pipeline code I've found nothing outstanding, what would block release.
